dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -2207,6 +2236,42 @@ private BulkActionsResultView distributeWorkAndProcess(final WorkflowAction acti
 	}
 
 
+	/**
+	 * This method takes the all the Batch actions saved in the batchActionContext and executes them
+	 * Also gathers any exception the might result of an internal error while processing the actions getting executed as a single batch.
+	 * @param user
+	 * @param action
+	 * @param actionsContext
+	 * @param failsConsumer
+	 * @throws DotDataException
+	 */
+	private void executeBatchActions(
+			final User user,
+			final WorkflowAction action,
+			final ConcurrentMap<String, Object> actionsContext,
+			final BiConsumer<List<?>, Exception> failsConsumer) throws DotDataException {
+
+		final List<WorkflowActionClass> actionClasses = this.findActionClasses(action);
+		for (final WorkflowActionClass actionClass : actionClasses) {
+			final WorkFlowActionlet actionlet = actionClass.getActionlet();
+			if (actionlet instanceof BatchAction) {
+				final Map<String, WorkflowActionClassParameter> params = findParamsForActionClass(actionClass);
+				final BatchAction batchAction = BatchAction.class.cast(actionlet);
+				final List <?> objects = batchAction.getObjectsForBatch(actionsContext, actionClass);
+				try {
+					batchAction.executeBatchAction(user, actionsContext, actionClass, params);
+				} catch (Exception e) {
+					failsConsumer.accept(objects, e);

```

Comment:
If a batch action fails should we continue with the rest of batch actions?

Reply:
I think not. 
They are inserted internally using this method :
publisherAPI.addContentsToPublish(identifiers, bundle.getId(), publishDate, user);

It not very likely that such thing is going to fail. But if it happens I think the whole batch will be rolled back since internally that method is enclosed within a @WrappedInTransaction